### PR TITLE
Improve the rounding off logic in Balance class

### PR DESCRIPTION
**Pull request by aniketsingh03**
### Changes
- This PR improves the logic in `misc.go` for improved rounding off when converting floats to ints 

### Fixes
- #752 

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli: https://github.com/0chain/zboxcli/pull/360
- zwalletcli: https://github.com/0chain/zwalletcli/pull/211
- Other: ...

### CODE REVIEW

File: core/common/misc.go [hunk 1]
```diff
@@ -2,10 +2,30 @@ package common
...
+	// ErrNegativeValue is returned if a float value is a negative number
+	ErrNegativeValue = errors.New("", "negative coin value")
+	// ErrTooManyDecimals is returned if a value has more than 10 decimal places
+	ErrTooManyDecimals = errors.New("", "too many decimal places")
+	// ErrTooLarge is returned if a value is greater than math.MaxInt64
+	ErrTooLarge = errors.New("", "value is too large")
+	// ErrUint64OverflowsFloat64 is returned if when converting a uint64 to a float64 overflow float64
+	ErrUint64OverflowsFloat64 = errors.New("", "uint64 overflows float64")
+	// ErrUint64AddOverflow is returned if when adding uint64 values overflow uint64
+	ErrUint64AddOverflow = errors.New("", "uint64 addition overflow")
```
Comment (peterlimg) [inline core/common/misc.go]: Let's use standard errors package, so `errors.New("uint64 addition overflow")`, so as the other errors above.

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix-rounding-off-for-balance  |
| zboxcli  | fix-rounding-off-for-balance  |
| zwalletcli  | fix-rounding-off-for-balance  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by peterlimg — APPROVED** [review]
LGTM

**Review by cnlangzi — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | fix-rounding-off-for-balance  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**